pytealutils/iter/iter.py

```diff
@@ -0,0 +1,15 @@
+from pyteal import Int, Expr, Subroutine, ScratchVar, For, TealType
+
+# TODO: pass the scratch var to use?
+
+# TODO: union type?
+def range(n: Int, sub: Expr):
```

Comment:
`range` has very a strong connotation in Python and is different from what this function does in PyTEAL. Also, this is a simplified version of the general PyTEAL `For()`. I can't think of a great name, maybe `basic_for()` or `iterate()`. Or maybe, just modify the PyTEAL `class For:` to have a factory method such as `iterate(iterations: int, sub: Expr)` which does the same thing.

Reply:
You can also subclass `For` here and add the `iterate` method. So something like:

```python
class Range(For):
    def __init__(self, iterations: int):
        ... call `super` with the necessary params ...

    @classmethod
    def iterate(cls, iterations: int, sub: Expr):
         return cls(iterations).Do(sub)
```

then you can run it with syntax such as:

```python
Range.iterate(iterations, sub)
```